src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs

```diff
@@ -158,5 +159,237 @@ public static class PermutationFeatureImportanceExtensions
+        /// </para>
+        /// <para>
+        /// In this implementation, PFI computes the change in all possible ranking evaluation metrics for each feature, and an
+        /// <code>ImmutableArray</code> of <code>RankingMetrics</code> objects is returned. See the sample below for an
+        /// example of working with these results to analyze the feature importance of a model.
+        /// </para>
+        /// </remarks>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[PFI](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/PermutationFeatureImportance.cs)]
+        /// ]]>
+        /// </format>
+        /// </example>
+        /// <param name="ctx">The clustering context.</param>
+        /// <param name="model">The model to evaluate.</param>
+        /// <param name="data">The evaluation data set.</param>
+        /// <param name="label">Label column name.</param>
+        /// <param name="groupId">GroupId column name</param>
+        /// <param name="features">Feature column names.</param>
+        /// <param name="useFeatureWeightFilter">Use features weight to pre-filter features.</param>
+        /// <param name="topExamples">Limit the number of examples to evaluate on. null means examples (up to ~ 2 bln) from input will be used.</param>
+        /// <returns>Array of per-feature 'contributions' to the score.</returns>
+        public static ImmutableArray<RankerMetrics>
+            PermutationFeatureImportance(
+                this RankingContext ctx,
+                IPredictionTransformer<IPredictor> model,
+                IDataView data,
+                string label = DefaultColumnNames.Label,
+                string groupId = DefaultColumnNames.GroupId,
+                string features = DefaultColumnNames.Features,
+                bool useFeatureWeightFilter = false,
+                int? topExamples = null)
+        {
+            return PermutationFeatureImportance<RankerMetrics>.GetImportanceMetricsMatrix(
+                            CatalogUtils.GetEnvironment(ctx),
+                            model,
+                            data,
+                            idv => ctx.Evaluate(idv, label, groupId),
+                            RankingDelta,
+                            features,
+                            useFeatureWeightFilter,
+                            topExamples);
+        }
+
+        private static RankerMetrics RankingDelta(
+            RankerMetrics a, RankerMetrics b)
+        {
+            if (a.Dcg.Length!= b.Dcg.Length || a.Ndcg.Length != b.Ndcg.Length)
+                throw new ArgumentException($"Can't compare {nameof(RankerMetrics)} with different DCG/NDCG Lengths.");
+
+            var dcg = new double[a.Dcg.Length];
+            for (int i = 0; i < a.Dcg.Length; i++)
+                dcg[i] = a.Dcg[i] - b.Dcg[i];
```

Comment:
Since we have 3 of these Array_A - Array_B operations, it would be cleaner to define it as a function or Array extension method.

Reply:
Done.